pkg/odp/config.go

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+ * Copyright 2022, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package odp //
+package odp
+
+import (
+	"sync"
+
+	"github.com/optimizely/go-sdk/pkg/odp/utils"
+)
+
+// Config is used to represent odp config
+type Config interface {
+	Update(apiKey, apiHost string, segmentsToCheck []string) bool
+	GetAPIKey() string
+	GetAPIHost() string
+	GetSegmentsToCheck() []string
+	IsOdpServiceIntegrated() bool
+}
+
+// DefaultConfig represents default implementation of odp config
+type DefaultConfig struct {
```

Comment:
i need to check name.

Reply:
couldn't name this `Config` as it was the name of the interface, also `ODPConfig` is not recommended by linter since the config resides under `odp` package. Any suggestions?